R/check-invalid-characters.R

```diff
@@ -0,0 +1,76 @@
+#' @title Check for non-ascii characters
+#'
+#' @description Check for non-ascii characters in columns.
+#'
+#' @param data Data to check
+#' @inheritParams check_values
+#' @return A condition object indicating whether the data contains columns with
+#' a non-ascii character.
+#' @export
+#' @examples
+#' dat <- tibble::tibble(
+#'   fails1 = c("study 1", "study&amp;2"),
+#'   succeeds = c("file1.ext", "file2.ext"),
+#'   fails2 = c("foo<0xa0>", "bar")
+#' )
+#' check_invalid_characters(dat)
+check_invalid_characters <- function(data,
+                             success_msg = "There are no invalid characters",
+                             fail_msg = "There is an invalid character in a column") { #nolint
+  if (is.null(data)) {
+    return(NULL)
+  }
+  has_invalid <- purrr::map_lgl(data, ~ contains_invalid(.))
+  behavior <- glue::glue(
+    "Only standard ascii characters are allowed."
```

Comment:
Where does `behavior` get surfaced? I'm not seeing this message, and it would be helpful to know why the characters are invalid.

Reply:
That behavior isn't surfaced in the current version of the app. Ideally, someone could add this to the `check_all` function instead of having it block upload (which is what I'm using it for in this version), which is why I made it in a similar format to the other types of validation checks. The downside to using it in `check_all` and having the result pop up in the results boxes is that other pieces of the app (summary/details tab and other validation checks) end up breaking, as well.

I added a bit of text to the error summary (it's updated on the test version). Now it should say:
```
Error: Only standard ascii characters are allowed in the files.
There are invalid characters in the manifest columns: study
There are invalid characters in the individual metadata columns: individualID, sex
```

How is that, @kelshmo?

Reply:
That will definitely be helpful. I will say I appreciate the error message identifying which file and column have non-standard characters!